zproject/prod_settings_template.py

```diff
@@ -304,6 +304,15 @@
 # SSO_APPEND_DOMAIN = "example.com")
 SSO_APPEND_DOMAIN = None  # type: Optional[str]
 
+# The name of the HTTP header if authentication is from a remote proxy
+# SSO_HTTP_USER_HEADER = "X-Remote-User"
+
+# The name of the HTTP header containing a shared secret between
+# the authenticating proxy and Zulip.
+# If this is set you must also define remote_sso_secret with the
+# shared secret in your zulip-secrets.conf
+# SSO_HTTP_SECRET_HEADER = "X-Remote-Secret"
```

Comment:
We might want to pick a better name prefix for these settings before adding more of them.  Maybe 

`REMOTE_USER_AUTH_USER_HEADER`
`REMOTE_USER_AUTH_SECRET_HEADER`
`REMOTE_USER_AUTH_DOMAIN`

(And then we'll do some backwards-compatibility logic for `SSO_APPEND_DOMAIN`).